# NYC Flights 2013 Analysis

In [ ]:
install.packages("dplyr")

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [ ]:
library("dplyr")


Attaching package: ‘dplyr’


The following objects are masked from ‘package:stats’:

    filter, lag


The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union




In [ ]:
flights <- read.csv("flights.csv")

In [ ]:
flights %>% 
  head(5)

year month day dep_time sched_dep_time dep_delay arr_time sched_arr_time
1 2013 1     1   517      515             2         830      819          
2 2013 1     1   533      529             4         850      830          
3 2013 1     1   542      540             2         923      850          
4 2013 1     1   544      545            -1        1004     1022          
5 2013 1     1   554      600            -6         812      837          
  arr_delay carrier flight tailnum origin dest air_time distance hour minute
1  11       UA      1545   N14228  EWR    IAH  227      1400     5    15    
2  20       UA      1714   N24211  LGA    IAH  227      1416     5    29    
3  33       AA      1141   N619AA  JFK    MIA  160      1089     5    40    
4 -18       B6       725   N804JB  JFK    BQN  183      1576     5    45    
5 -25       DL       461   N668DN  LGA    ATL  116       762     6     0    
  time_hour           
1 2013-01-01T10:00:00Z
2 2013-01-01T10:00:00Z
3 2013-01-01T10:00:00Z
4 2013-01-01T10:00:00Z
5 2013-01-01T11:00:00Z

### Q1: 5 อันดับสายการบินที่เดินทางมากที่สุดในเดือนมิถุนายน 2013

In [ ]:
flights %>%
  filter(month == 6, year == 2013) %>%
  count(carrier) %>%
  arrange(desc(n)) %>%
  head(5)

carrier n   
1 UA      4975
2 B6      4622
3 EV      4456
4 DL      4126
5 AA      2757

จากข้อมูลพบว่าสายการบินที่เดินทางมากที่สุดคือUA อันดับสองคือB6

### Q2: 5 อันดับสายการบินที่มีจำนวนครั้งล่าช้าในการออกเดินทางมากที่สุด


In [ ]:
flights %>%
    select(dep_delay, carrier) %>%
    filter(dep_delay > 0) %>%
    count(carrier) %>%
    arrange(desc(n)) %>%
  head(5)

carrier n    
1 UA      27261
2 EV      23139
3 B6      21445
4 DL      15241
5 AA      10162

จากข้อมูลพบว่าสายการบินUAล่าช้าไปทั้งหมด 27261 ครั้ง อันดับ2คือEV 23139ครั้ง

### Q3: 5 อันดับสายการบินที่มีจำนวนเวลาในการออกเดินทางล้าช้าสะสมมากที่สุด

In [ ]:
flights %>%
    select(dep_delay, carrier) %>%
    filter(dep_delay > 0) %>%
    group_by(carrier) %>%
    summarise(delay = round(sum(dep_delay)/60,digits = 1)) %>%
    arrange(desc(delay)) %>%
  head(5)


carrier delay  
1 EV      19409.7
2 B6      14223.1
3 UA      13597.0
4 DL       9500.3
5 AA       6295.2

จากข้อมูลพบว่าสายการบินEV มีจำนวนชม.ของการล่าช้าสะสมมากที่สุด

### Q4: สายการบินที่คนนิยมใช้บริการมากที่สุดในปี โดยคิดจากจำนวนการบิน

In [ ]:
flights %>%
  count(carrier) %>%
  arrange(desc(n)) %>% 
  head(10)

carrier n    
1  UA      58665
2  B6      54635
3  EV      54173
4  DL      48110
5  AA      32729
6  MQ      26397
7  US      20536
8  9E      18460
9  WN      12275
10 VX       5162

ในปี2013 สายการบินUAเป็นที่นิยมมากที่สุด

### Q5: จำนวนสายการบินที่มีระยะเดินทางสะสมมากที่สุดในปี 2013

In [ ]:
flights %>%
  select(distance, carrier) %>%
  group_by(carrier) %>%
  summarise(sum_distance = sum(distance)) %>%
  arrange(desc(sum_distance))
  

carrier sum_distance
1  UA      89705524    
2  DL      59507317    
3  B6      58384137    
4  AA      43864584    
5  EV      30498951    
6  MQ      15033955    
7  VX      12902327    
8  WN      12229203    
9  US      11365778    
10 9E       9788152    
11 FL       2167344    
12 AS       1715028    
13 HA       1704186    
14 F9       1109700    
15 YV        225395    
16 OO         16026

### Q6: รุ่นเครื่องบินที่ถูกใช้งานมากที่สุด

In [ ]:
flights %>%
  filter(!is.na(tailnum)) %>%
  count(tailnum) %>%
  arrange(desc(n)) %>% 
  head(10)

tailnum n  
1  N725MQ  575
2  N722MQ  513
3  N723MQ  507
4  N711MQ  486
5  N713MQ  483
6  N258JB  427
7  N298JB  407
8  N353JB  404
9  N351JB  402
10 N735MQ  396

ในปี2013 พบว่าเครื่องบินรุ่น N725MQ ถูกนำมาใช้งานมากที่สุด รองลงมาคือN722MQและN723MQ

Q7: สนามบินต้นทางที่คนเข้าไปใช้บริการมากที่สุด

In [ ]:
flights %>%
  count(origin) %>%
  arrange(desc(n))

origin n     
1 EWR    120835
2 JFK    111279
3 LGA    104662

### Q8: สนามบินปลายทางที่คนนิยมไป 20 อันดับ

In [ ]:
flights %>%
  count(dest) %>%
  arrange(desc(n))%>%
  head(20)

dest n    
1  ORD  17283
2  ATL  17215
3  LAX  16174
4  BOS  15508
5  MCO  14082
6  CLT  14064
7  SFO  13331
8  FLL  12055
9  MIA  11728
10 DCA   9705
11 DTW   9384
12 DFW   8738
13 RDU   8163
14 TPA   7466
15 DEN   7266
16 IAH   7198
17 MSP   7185
18 PBI   6554
19 BNA   6333
20 LAS   5997

จากข้อมูลพบว่าปลายทางที่คนนิยมเดินทางไปมากที่สุดคือORD ลองลงมาคือATL

### Q9: คนที่ไปORDนิยมใช้บริการสายการบินอะไร


In [ ]:
flights %>%
  select(dest,carrier) %>%
  filter(grepl("ORD", flights$dest)) %>%
  count(carrier) %>%
  arrange(desc(n))
  

carrier n   
1 UA      6984
2 AA      6059
3 MQ      2276
4 9E      1056
5 B6       905
6 EV         2
7 OO         1

สายการบินที่เป็นที่นิมยมในการเดินทางไปORDคือUA รองลงมาคือ AA

### Q10: เดือนไหนที่คนไป ORD มากที่สุด

In [ ]:
flights %>%
  select(dest,month) %>%
  filter(grepl("ORD", flights$dest)) %>%
  count(month) %>%
  arrange(desc(n))

month n   
1   8    1604
2  10    1604
3   5    1582
4   9    1582
5   7    1573
6   6    1547
7   4    1416
8  11    1370
9   3    1343
10  1    1269
11  2    1197
12 12    1196

เดือนที่คนนิยมเดินทางไปORDคือเดือน8และเดือน10

# postgresql

In [ ]:
install.packages("RPostgreSQL")

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [ ]:
library(RPostgreSQL)

Loading required package: DBI



In [ ]:
con <- dbConnect(
  PostgreSQL(),
  host = "tiny.db.elephantsql.com",
  dbname = "suvqbsul",
  port = 5432,
  user = "suvqbsul",
  password = "YhlfaU6WT1zZos9Q6W5R0jBZaDiaEjvE"
)

In [ ]:
# df 1
students <- data.frame(
  id = 1:5,
  student_name = c("Pariwat","Sompong","Thitima","Umpai","James")
)

In [ ]:
# df 2
products <- data.frame(
  id = 1:5,
  name = c("Banana","Apple","Mango","Papaya","Kiwi"),
  price = c(100,200,300,400,500),
  quantity = c(200,200,200,200,200)
)

In [ ]:
# df 3
delivery <- data.frame(
  id = 1:5,
  product = c(1,1,3,4,5),
  zone = c(1,2,3,4,5),
  pay= c(TRUE,FALSE,TRUE,TRUE,FALSE)
)

In [ ]:
# df 4
zone <- data.frame(
  id = 1:5,
  name = c("DMK","BKK","HDY","CNX","CEI")
)

In [ ]:
dbWriteTable(con, "students", students, row.names = FALSE)

[1] TRUE

In [ ]:
dbWriteTable(con, "products", products, row.names = FALSE)

[1] TRUE

In [ ]:
dbWriteTable(con, "deliverys", delivery, row.names = FALSE)

[1] TRUE

In [ ]:
dbWriteTable(con, "zone", zone, row.names = FALSE)

[1] TRUE

In [ ]:
dbListTables(con)

[1] "students"  "products"  "zone"      "deliverys"

In [ ]:
dbGetQuery(con,"SELECT * FROM students")

id student_name
1 1  Pariwat     
2 2  Sompong     
3 3  Thitima     
4 4  Umpai       
5 5  James

In [ ]:
dbGetQuery(con,"SELECT 
pr.name,
pr.price,
pr.quantity,
zone.name as zone,
de.pay
FROM deliverys de
JOIN products pr
ON de.product = pr.id
JOIN zone 
ON de.zone = zone.id")

name   price quantity zone pay  
1 Banana 100   200      DMK   TRUE
2 Banana 100   200      BKK  FALSE
3 Mango  300   200      HDY   TRUE
4 Papaya 400   200      CNX   TRUE
5 Kiwi   500   200      CEI  FALSE

In [ ]:
dbDisconnect(con)

[1] TRUE